File: Experiment07Fragments/app/src/main/java/com/example/experiment07fragments/MainActivity.java
package com.example.experiment07fragments;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentTransaction;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.LinearLayout;

public class MainActivity extends AppCompatActivity {

    Button button1,button2;
    LinearLayout linearLayout;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        button1 = findViewById(R.id.button1);
        button2 = findViewById(R.id.button2);
        linearLayout = findViewById(R.id.linearlayout);
//        button1.setOnClickListener(new View.OnClickListener() {
//            @Override
//            public void onClick(View v) {
//                Fragment1 firstFragment = new Fragment1();
//                FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
//                transaction.replace(R.id.linearlayout,firstFragment);
//                transaction.commit();
//            }
//        });

        button1.setOnClickListener(view ->{
            Fragment1 firstFragment = new Fragment1();
            FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
            transaction.replace(R.id.linearlayout,firstFragment);
            transaction.commit();
        });
        button2.setOnClickListener(view->{
            Fragment2 secondFragment = new Fragment2();
            FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
            fragmentTransaction.replace(R.id.linearlayout,secondFragment);
            fragmentTransaction.commit();
        });
    }
}

File: DialogBox/app/src/main/java/com/example/dialogbox/MainActivity.kt
package com.example.dialogbox

import android.app.AlertDialog
import android.app.Dialog
import android.app.ProgressDialog
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        lateinit var btn:Button
        lateinit var btn2:Button
        var builder=AlertDialog.Builder(this)
       val progressDialog = ProgressDialog(this)

        btn=findViewById(R.id.click)
        btn2=findViewById(R.id.progress)
        btn.setOnClickListener{
            builder.setTitle("AlertBox")
            builder.setMessage("Are you sure you want to exit")
            builder.setPositiveButton("yes"){dialog,which->
                Toast.makeText(this,"Thanks for using ",Toast.LENGTH_SHORT).show()

                finish()
        }
            builder.setNegativeButton("No"){dialog,which->
                Toast.makeText(this,"Activity Resume",Toast.LENGTH_SHORT).show()
            }
            builder.show()
        }
        btn2.setOnClickListener{
            progressDialog.setTitle("Wait for response")
            progressDialog.setMessage("Just a moment")
            progressDialog.show()
        }
    }
}